hazelcast/src/main/java/com/hazelcast/instance/NodeExtension.java

```diff
@@ -130,7 +131,7 @@
     /**
      * Creates factory method that creates server side client message handlers
      */
-    MessageTaskFactory createMessageTaskFactory();
+    List<? extends MessageTaskFactory> createMessageTaskFactories();
```

Comment:
List is needed because we can have many factories for component.
Some factories are being provided by Hazelcast and by Jet